### DISCUSSION TITLE
Is the "ignore comments" option to json::parse working properly?

### DISCUSSION THREAD

**Discussion by bbalaban** [nlohmann/json#4113] [Q&A] [answered]
I'm trying out the optional "ignore comments" argument to json::parse(), and it doesn't seem to be working -- it's discarding the json.
Here's an example of the test JSON i'm trying to parse:
   {
        "@unid": "B2D6958DCEF3B003C12588B5003719CD",	// unid
        "@noteid": 4382,				// note id
        "@index": "32.27",				// index
        "$6": "1.0",					// no idea what this is
        "gender": "Female",				// gender
        "city": "New York City",			// where?
        "last_name": "Dovidian",			// family name
        "state": "New York",				// where?
        "first_name": "Florrie",			// given name
        "email": "fdovidianei@deliciousdays.com"	// email
    }

and my calling code looks like this:
    json j = json::parse(bytes,
        NULL,       // no callback
        false,  	// no exceptions please
        true);      // allow comments

Without the comments in the input JSON it works fine, with the comments, i get a "discarded" result. 
I'm on version 3.9.1
Am I doing something wrong?

**Comment by nlohmann** [comment]
If you enable exceptions, what is the error message?

**Comment by bbalaban** [comment]
I enabled exceptions, got the same result. Returned json object is either
null or discarded

On Wed, Aug 16, 2023 at 2:18 PM Niels Lohmann ***@***.***>
wrote:

> If you enable exceptions, what is the error message?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nlohmann/json/discussions/4113#discussioncomment-6743730>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEJS6VDD7ZBCK2RTNSN66S3XVUFHBANCNFSM6AAAAAA3S3CSPY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Find Bob's blog at http://www.bobzblog.com

**Reply by nlohmann** [reply]
That cannot happen. Can you please share your complete code?

**Reply by bbalaban** [reply]
json j = json::parse(bytes,
        NULL,       // no callback
        true,  	// enable exceptions
        true);      // allow comments

    delete [] bytes;

    if (!j.is_null() && !j.is_discarded())
        this->m_json = j;
    else {
        //....
       // raise error and abort
        return;
    }

Is it possible that different line endings after the comments would affect the parsing? E.g., \n vs. \r\n?

**Reply by bbalaban** [reply, marked as answer]
I figured it out, the code that was reading the text from a disk file was filtering out newlines, so ... own-goal
